dblookupext/esdtSupply/logsProcessor.go

```diff
@@ -0,0 +1,149 @@
+package esdtSupply
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type logsProcessor struct {
+	marshalizer        marshal.Marshalizer
+	suppliesStorer     storage.Storer
+	fungibleOperations map[string]struct{}
+}
+
+func newLogsProcessor(
+	marshalizer marshal.Marshalizer,
+	suppliesStorer storage.Storer,
+) *logsProcessor {
+	return &logsProcessor{
+		marshalizer:    marshalizer,
+		suppliesStorer: suppliesStorer,
+		fungibleOperations: map[string]struct{}{
+			core.BuiltInFunctionESDTLocalBurn:      {},
+			core.BuiltInFunctionESDTLocalMint:      {},
+			core.BuiltInFunctionESDTNFTCreate:      {},
+			core.BuiltInFunctionESDTNFTAddQuantity: {},
+			core.BuiltInFunctionESDTNFTBurn:        {},
+		},
+	}
+}
+
+func (lp *logsProcessor) processLogs(logs map[string]data.LogHandler, isRevert bool) error {
+	supplies := make(map[string]*SupplyESDT)
+	for _, logHandler := range logs {
+		if check.IfNil(logHandler) {
+			continue
+		}
+
+		err := lp.processLog(logHandler, supplies, isRevert)
+		if err != nil {
+			return err
+		}
+	}
+
+	return lp.saveSupplies(supplies)
+}
+
+func (lp *logsProcessor) processLog(txLog data.LogHandler, supplies map[string]*SupplyESDT, isRevert bool) error {
+	for _, entryHandler := range txLog.GetLogEvents() {
+		if check.IfNil(entryHandler) {
+			continue
+		}
+
+		event, ok := entryHandler.(*transaction.Event)
+		if !ok {
+			continue
+		}
+
+		if lp.shouldIgnoreEvent(event) {
+			continue
+		}
+
+		err := lp.processEvent(event, supplies, isRevert)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (lp *logsProcessor) saveSupplies(supplies map[string]*SupplyESDT) error {
+	for identifier, supplyESDT := range supplies {
+		supplyESDTBytes, err := lp.marshalizer.Marshal(supplyESDT)
+		if err != nil {
+			return err
+		}
+
+		err = lp.suppliesStorer.Put([]byte(identifier), supplyESDTBytes)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (lp *logsProcessor) processEvent(txLog *transaction.Event, supplies map[string]*SupplyESDT, isRevert bool) error {
+	tokenIdentifier := txLog.Topics[0]
```

Comment:
add check for length of txLog.Topics - this can panic

Reply:
added extra check.